### PR TITLE
Add flow typings

### PR DESCRIPTION
**Pull request by andreypopp**
Few questions/notes:

* I put flow typing declarations within `dist/` directory. This is iconsistent with the location of typescript declarations as for now. This is because typescripts makes typings location configurable through the `package.json` while Flow can't do this.

* I added some tests for Flow typings (they are minimal now, will expand as I accumulate more usage patterns) to `test_flow`. Test are run by `npm test` command so CI can check them too. Not sure if this is the best location / integration with CI. What do you think?

Related to #40.

### CODE REVIEW

File: dist/derivable.js.flow [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  not(): Derivable<boolean>;
+
+  switch(...args: Array<mixed>): Derivable<mixed>;
+
+  withEquality(equals: (a: mixed, b: mixed) => boolean): Derivable<T>;
+};
+
+export interface Atom<T> extends Derivable<T> {
+
+  set(value: T): void;
```
Comment (andreypopp) [inline dist/derivable.js.flow]: I made `set()` not to change type parameter of the atom. I think this more aligns with how it's going to be used. Also Flow could infer type of:
```
let a = atom(42);
a.set('ok')
```
as `Atom<string | number>` automatically (yay control flow analysis!) automatically!

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
   ],
   "scripts": {
     "build": "node scripts/build.js",
-    "test": "mocha --recursive",
+    "test": "mocha --recursive && (cd test_flow; npm install; npm test)",
```
Comment (andreypopp) [inline package.json]: Not sure this is the best setup waiting for your comments.

File: package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   ],
   "scripts": {
     "build": "node scripts/build.js",
-    "test": "mocha --recursive",
+    "test": "mocha --recursive && (cd test_flow && npm install && npm test)",
```
Comment (andreypopp) [inline package.json]: Not sure this is the best setup waiting for your comments.

Comment (ds300) [inline package.json]: :/

It's not ideal since I often pass extra args to `npm test` when I'm debugging. I could come up with a workaround, though.

**Comment by ds300** [comment]
Thanks :) This looks nice.

One problem is that `dist/` gets blown away by the build process, so we'd have to copy the typings in from elsewhere as a `postinstall` step.

Also, can you add test_flow to `.npmignore`?

**Comment by andreypopp** [comment]
> so we'd have to copy the typings in from elsewhere as a postinstall step.

Do you want me to do the same for typescript typings? Just for consistency.

File: package.json (hunk 3 above)
Comment (andreypopp) [inline package.json]: What about adding `ci:test` script and configuring Travis to run that instead?

**Comment by ds300** [comment]
Ok fine. Not really bothered either way, as long as everything works.

File: package.json [hunk 4]
```diff
@@ -49,7 +49,9 @@
     "mobx": "^2.3.3",
     "mocha": "^2.2.5",
     "mocha-istanbul": "^0.2.0",
+    "promise": "^7.1.1",
     "rollup": "^0.26.1",
+    "source-map": "^0.5.6",
```
Comment (andreypopp) [inline package.json]: This was missing but used in `scripts/build.js`.

File: package.json [hunk 5]
```diff
@@ -49,7 +49,9 @@
     "mobx": "^2.3.3",
     "mocha": "^2.2.5",
     "mocha-istanbul": "^0.2.0",
+    "promise": "^7.1.1",
```
Comment (andreypopp) [inline package.json]: This was missing but used in `scripts/build.js`.

**Comment by andreypopp** [comment]
Travis now fails because I didn't commit the build artifacts. (btw. is there a need to keep them commited to the repo?).

**Comment by ds300** [comment]
I suppose not. We could just run `npm build` as part of the postinstall.

**Comment by andreypopp** [comment]
> We could just run npm build as part of the post install.

I think `npm` does that for you.

**Comment by andreypopp** [comment]
Nope, of course I'm wrong, npm does `build` on publish not on `postinstall`. Making it on `postinstall` will require to have all dev deps installed which isn't great. I suggest just removing build artifacts and thus breaking install directly from github repo.

**Comment by andreypopp** [comment]
btw. I added `ci:test` and configured travis to run it.

**Comment by andreypopp** [comment]
Added `npm run build` to Travis (which is also cool as it checks the build pipeline) so it's green now.

**Comment by ds300** [comment]
Thanks! 🎉

**Comment by andreypopp** [comment]
Thank you! This all is shaping up so well!

**Pull request merged**